### PR TITLE
debug_traceCallMany: strip filler blocks from overrides result

### PR DESCRIPTION
**Pull request by grubbyTimes**
`debug_traceCallMany` overrides path returned traces for every simulated block, including the filler blocks `SimulateTxExecutor` inserts when `BlockOverride.Number` has gaps. Result count no longer matched the input bundles.

Compute each bundle's target block number (same `overrideNumber ?? lastBlockNumber + 1` rule the executor uses) and filter `simulationResult.Data` down to that set. Regression test covers a non-contiguous gap between two bundles.

### CODE REVIEW

**Comment by LukaszRozmej** [comment]
@claude review

**Comment by grubbyTimes** [comment]
Added the gap=1 boundary case for "Test gap"; skipping "Logic coupling" (extracting a helper just relocates the duplication unlessSimulateTxExecutor also calls it — better as a separate IsFiller refactor PR) and "No completeness check" (Debug.Assert compiles out in Release, so it wouldn't guard prod).

**Comment by svlachakis** [comment]
@claude review

File: src/Nethermind/Nethermind.JsonRpc/Modules/DebugModule/DebugRpcModule.cs [hunk 1]
```diff
@@ -482,6 +482,19 @@ private ResultWrapper<IEnumerable<IEnumerable<GethLikeTxTrace>>> TraceCallManyWi
...
 
+        // SimulateTxExecutor inserts filler blocks between bundles when BlockOverride.Number has gaps.
+        // Pre-compute the block number each bundle targets so we can drop fillers from the result and
+        // keep a 1:1 mapping to the input bundles.
+        HashSet<long> bundleBlockNumbers = new(bundles.Length);
+        long lastBlockNumber = header.Number;
+        foreach (TransactionBundle bundle in bundles)
+        {
+            ulong? overrideNumber = bundle.BlockOverride?.Number;
+            long number = overrideNumber is not null ? (long)overrideNumber.Value : lastBlockNumber + 1;
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.JsonRpc/Modules/DebugModule/DebugRpcModule.cs]: maybe collapse the formula into helper method and reuse it in the Executor too?

```suggestion
            long number = bundle.BlockOverride.GetBlockNumber(lastBlockNumber);
```

where `GetBlockNumber` is an extension method that works with null's too

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Nethermind.JsonRpc/Modules/DebugModule/DebugRpcModule.cs (hunk 1 above)
Comment (grubbyTimes) [reply src/Nethermind/Nethermind.JsonRpc/Modules/DebugModule/DebugRpcModule.cs]: Sure nice to have

**Review by LukaszRozmej — APPROVED** [review]

**Review by batrr — APPROVED** [review]

**Review by svlachakis — APPROVED** [review]
